runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -782,8 +787,13 @@ private void checkVersion() {
                             getVersionString(), version.getVersion());
                 }
             }
-        } catch (TimeoutException | NetworkException | ShutdownException e) {
-            log.error("connect: failed to get version", e);
+        } catch (TimeoutException | NetworkException | ShutdownException | UnavailableServerException e) {
+            log.error("connect: failed to get version. Couldn't connect to server.", e);
+        } catch (Exception ex) {
+            // Because checkVersion is just an informational step (log purpose), we don't need to retry
```

Comment:
These comments apply to the other catch clause as well. Maybe bring put them before both or in the comment for the method. Also the only difference between the catch clauses is a sentence in the log message (e.g. **Couldn't connect to server.**). If it is not critical combine them to simplify.